picoscope/ps4000a.py

```diff
@@ -277,29 +285,77 @@ def _lowLevelGetTimebase(self, tb, noSamples, oversample, segmentIndex):
-
-        if sampleTimeS <= 12.5E-9:
-            timebase = math.floor(math.log(sampleTimeS * 8E7, 2))
-            timebase = max(timebase, 0)
-        else:
-            # Otherwise in range 2^32-1
-            if sampleTimeS > maxSampleTime:
-                sampleTimeS = maxSampleTime
+        """ Convert the sample interval (float of seconds) to the
+        corresponding integer timebase value as defined by the API.
+        See "Timebases" section of the PS4000a programmers guide
+        for more information.
+        """
 
-            timebase = math.floor((sampleTimeS * 2e7) + 1)
+        if self.model == '4828':
+            maxSampleTime = (((2 ** 32 - 1) + 1) / 8E7)
+
+            if sampleTimeS <= 12.5E-9:
+                timebase = 0
+            else:
+                # Otherwise in range 2^32-1
+                if sampleTimeS > maxSampleTime:
+                    sampleTimeS = maxSampleTime
+
+                timebase = math.floor((sampleTimeS * 2e7) + 1)
+
+        elif self.model == '4444':
+            maxSampleTime = (((2 ** 32 - 1) - 2) / 5.0E7)
+
+            if (sampleTimeS <= 2.5E-9 and
+                    self.resolution == self.ADC_RESOLUTIONS["12"]):
+                timebase = 0
+            elif (sampleTimeS <= 20E-9 and
+                    self.resolution == self.ADC_RESOLUTIONS["14"]):
+                timebase = 3
+            else:
+                # Otherwise in range 2^32-1
+                if sampleTimeS > maxSampleTime:
+                    sampleTimeS = maxSampleTime
+
+                timebase = math.floor((sampleTimeS * 5.0E7) + 2)
+
+        else:  # The original case from non "A" series
+            warnings.warn(message="The model PS4000a you are using may not be"
+                                  "fully supported", stacklevel=2)
+            maxSampleTime = (((2 ** 32 - 1) - 4) / 2e7)
+
+            if sampleTimeS <= 12.5E-9:
+                timebase = math.floor(math.log(sampleTimeS * 8E7, 2))
+                timebase = max(timebase, 0)
+            else:
+                # Otherwise in range 2^32-1
+                if sampleTimeS > maxSampleTime:
+                    sampleTimeS = maxSampleTime
+
+                timebase = math.floor((sampleTimeS * 2e7) + 1)
 
         # is this cast needed?
         timebase = int(timebase)
         return timebase
 
     def getTimestepFromTimebase(self, timebase):
         """Return timebase to sampletime as seconds."""
-        if timebase < 3:
-            dt = 2. ** timebase / 8e7
-        else:
-            dt = (timebase - 1) / 2e7
+        if self.model == '4828':
+            dt = (timebase + 1) / 8.0E7
+        elif self.model == '4444':
+            if timebase < 3:
+                dt = 2.5 ** timebase / 4.0E8
+            else:
+                dt = (timebase - 2) / 5.0E7
+
+        else:  # The original case from non "A" series
+            warnings.warn(message="The model PS4000a you are using may not be"
```

Comment:
there needs to be a space after the "e" and before the quote.

If you feel like it, be as verbose as you want, you will thank me in the future ;)